initrd/bin/network-init-recovery

```diff
@@ -4,58 +4,104 @@
 
 TRACE_FUNC
 
-# bring up the ethernet; maybe should do DHCP?
+# bring up the ethernet interface
 ifconfig lo 127.0.0.1
 
+echo "Loading Ethernet network modules..."
 network_modules="e1000 e1000e igb sfc mdio mlx4_core mlx4_en"
-for module in `echo $network_modules`; do
+for module in $(echo $network_modules); do
 	if [ -f /lib/modules/$module.ko ]; then
-		insmod /lib/modules/$module.ko
+		#check if module is already loaded and load it if not
+		lsmod | grep -q $module || insmod /lib/modules/$module.ko
 	fi
 done
 
-if [ -e /sys/class/net/eth0 ]; then
+#Tethering over USB for Mobile phones supporting CDC (Android Pixel 6a+, Librem phone, etc.)
+if [ -e /lib/modules/cdc_ether.ko ]; then
+	#first enable USB controllers
+	enable_usb
+	echo ""
+	echo "Please verify that your mobile (CDC NCM/EEM tethering compatible phone) is networked in the desired way (WIFI/mobile + VPN/Orbot/etc)"
+	echo "Please connect mobile phone to this machine's fast USB port (blue identified) through a known working data cable"
+	echo "Please enable USB tethering prior of going further (Android: select 'Charging this device via USB' notification and enable tethering option)"
+	read -p "Press Enter to continue now or wait 30 seconds..." -n 1 -r -t 30
```

Comment:
@JonathonHall-Purism That may have been over-precautious, but that network-recovery-script was created as a "generic-init" alternative, which is not "known" to be used anymore because most boards switched to gui-init since whiptail replaced server board configs. But network-init-recovery, if not automated to timeout and simply continue, would fail launching dropbear if enabled in that script for server boards, IF USED (I do not know). The goal of 30 seconds here was to not break things as of now, and have issues report opened if any. As of now, I do not expect any issue being opened because it would not make sense that tethering is activated for headless systems.

Thoughts?
@JonathonHall-Purism If you insist, I can remove it, but as of now this is implementation detail. User with tethering will press Enter within 30 seconds.

Reply:
Thanks for the background, I still think we should remove it.

I understand the context, but in the simplest terms I think this is a compromise that satisfies no one, so IMO please remove it so we can push toward consistent UX :slightly_smiling_face: 

If somebody is using a modified build with network-init-recovery as the init script, it's still going to delay 60 seconds before dropbear comes up.  That's long enough to think the system is hosed if it's headless.  If this is something we need to support upstream (and I'm not saying it is :slightly_smiling_face: ), then I think it should be supported properly - the feature needs to properly exist somewhere (i.e. a config to make this your init script) and then separate the interactive functionality here somehow.  But since this feature doesn't exist in upstream, I don't think we should complicate our lives supporting a hypothetical feature.